src/kiwixchoicebox.cpp

```diff
@@ -204,10 +197,9 @@ void KiwixChoiceBox::clearSelections()
 {
     for (auto &item : choiceSelector->selectedItems()) {
         item->setSelected(false);
-        auto chItem  = m_choiceItems[item->text()];
-        currentChoicesLayout->removeWidget(chItem);
-        chItem->deleteLater();
-        m_choiceItems.remove(item->text());
+        auto chItem  = ui->currentChoices->findChild<ChoiceItem*>(item->text());
+        ui->currentChoices->layout()->removeWidget(chItem);
+        delete chItem;
```

Comment:
`chItem->deleteLater()` ?

Reply:
I had started with a `deleteLater` early on but seems like it doesn't delete just then. Later on, adding selection creates duplicates if the old aren't deleted, hence the `delete` usage.
It's also not inside a slot and sort of "independent", so I think it's safe to use delete.